waitforx/waitforx.c

```diff
@@ -0,0 +1,96 @@
+#include <X11/extensions/Xrandr.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/signal.h>
+#include <unistd.h>
+
+#define ATTEMPTS 10
+#define ALARM_WAIT 30
+
+void
+alarm_handler(int signal_num)
+{
+    printf("Unable to find RandR outputs after %d seconds\n", ALARM_WAIT);
+    exit(1);
+}
+
+int
+main(int argc, char **argv)
+{
+    char *display = NULL;
+    int error_base = 0;
+    int event_base = 0;
+    int n = 0;
+    int outputs = 0;
+    int wait = ATTEMPTS;
+
+    Display *dpy = NULL;
+    XRRScreenResources *res = NULL;
+
+    display = getenv("DISPLAY");
+
+    signal(SIGALRM, alarm_handler);
+    alarm(ALARM_WAIT);
```

Comment:
For consistency with the other signals these should probably be wrapped.

I'd suggest writing a function which does both the timer and setting the handler, as this could be moved to Windows later (not that this is likely to happen).

How does this look?

```c
unsigned int g_set_alarm(void (*func)(int), unsigned int secs);
```

```c
/*****************************************************************************/
/* does not work in win32 */
unsigned int
g_set_alarm(void (*func)(int), unsigned int secs)
{
#if defined(_WIN32)
    return 0;
#else
    /* Cancel any previous alarm to prevent a race */
    unsigned int rv = alarm(0);
    signal(SIGALRM, func);
    (void)alarm(secs);
    return rv;
#endif
}
```

Reply:
See 8c44ae0 and d7d0866